core/shared/src/main/scala/zio/Chunk.scala

```diff
@@ -807,7 +808,7 @@ object Chunk {
 
     override def toArray[A1 >: A](implicit tag: ClassTag[A1]): Array[A1] = array.asInstanceOf[Array[A1]]
 
-    override def length: Int = array.length
+    override val length: Int = array.length
```

Comment:
Another thing you can do: anytime `length` is referenced, instead put it into a local `val`. This way it won't go through `invokevirtual` like all (non-`private[this]`) `val` access does.